src/spyglass/decoding/v1/clusterless.py

```diff
@@ -0,0 +1,381 @@
+            classifier.initial_conditions_,
+            name="initial_conditions",
+        )
+        results["discrete_state_transitions"] = xr.DataArray(
+            classifier.discrete_state_transitions_,
+            dims=("states", "states"),
+            name="discrete_state_transitions",
+        )
+        if (
+            vars(classifier).get("discrete_transition_coefficients_")
+            is not None
+        ):
+            results[
+                "discrete_transition_coefficients"
+            ] = classifier.discrete_transition_coefficients_
+
+        # Insert results
+        # in future use https://github.com/rly/ndx-xarray and analysis nwb file?
+
+        nwb_file_name = key["nwb_file_name"].strip("_.nwb")
+
+        # Generate a unique path for the results file
+        path_exists = True
+        while path_exists:
+            results_path = (
+                Path(config["SPYGLASS_ANALYSIS_DIR"])
+                / nwb_file_name
+                / f"{nwb_file_name}_{str(uuid.uuid4())}.nc"
+            )
+            path_exists = results_path.exists()
+        classifier.save_results(
+            results,
+            results_path,
+        )
+        key["results_path"] = results_path
+
+        self.insert1(key)
+
+        from spyglass.decoding.decoding_merge import DecodingOutput
+
+        DecodingOutput.insert1(orig_key, skip_duplicates=True)
+
+    def load_results(self):
+        return ClusterlessDetector.load_results(self.fetch1("results_path"))
+
+    @staticmethod
+    def load_environments(key):
+        model_params = (
+            DecodingParameters
+            & {"decoding_param_name": key["decoding_param_name"]}
+        ).fetch1()
+        decoding_params, decoding_kwargs = (
+            model_params["decoding_params"],
+            model_params["decoding_kwargs"],
+        )
+
+        (
+            position_info,
+            position_variable_names,
+        ) = ClusterlessDecodingV1.load_position_info(key)
+        classifier = ClusterlessDetector(**decoding_params)
+
+        classifier.initialize_environments(
+            position=position_info[position_variable_names].to_numpy(),
+            environment_labels=decoding_kwargs.get("environment_labels", None),
+        )
+
+        return classifier.environments
+
+    @staticmethod
+    def load_position_info(key):
+        position_group_key = {"position_group_name": key["position_group_name"]}
+        position_variable_names = (PositionGroup & position_group_key).fetch1(
+            "position_variables"
+        )
+
+        position_info = []
+        for pos_merge_id in (PositionGroup.Position & position_group_key).fetch(
+            "pos_merge_id"
+        ):
+            position_info.append(
+                IntervalPositionInfo._data_to_df(
+                    PositionOutput.fetch_nwb({"merge_id": pos_merge_id})[0],
+                    prefix="",
+                    add_frame_ind=True,
```

Comment:
Is the frame index necessary for anything? It causes an error for upsampled position data

Reply:
Ah no it isn't

Reply:
We should probably just shift this to be the fetch1_dataframe function for PositionOutput anyways.